### PR TITLE
[SPARK-33014][SQL] Support multiple bucket columns in DataSourceV2 table

### PR DESCRIPTION
**Pull request by LantaoJin**
### What changes were proposed in this pull request?
In `V2SessionCatalog.convertTransforms()`, partitions only match one bucket column by
`case BucketTransform(numBuckets, FieldReference(Seq(col))) =>`
It should be a bug since there is no comment to explain the reason.
This PR accepts more than one bucket columns for V2 table.

### Why are the changes needed?
Create a data source V2 table only accept one bucket column, two or more bucket columns in V2 table will show exception `SessionCatalog does not support partition transform ...`.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Add UT.

### CODE REVIEW

**Comment by LantaoJin** [comment]
ping @cloud-fan

**Comment by LantaoJin** [comment]
gentle ping @cloud-fan @HyukjinKwon @dongjoon-hyun

**Comment by LantaoJin** [comment]
Gentle ping @cloud-fan @HyukjinKwon @dongjoon-hyun again

File: sql/catalyst/src/main/scala/org/apache/spark/sql/connector/expressions/expressions.scala [hunk 1]
```diff
@@ -115,11 +115,10 @@ private[sql] final case class BucketTransform(
 }
 
 private[sql] object BucketTransform {
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/connector/expressions/expressions.scala:117]: Looks like you should update DataFrameWriter.scala, functions.scala, functions.py, etc related codes. cc @rdblue, @brkyvz and @zero323 FYI

File: sql/catalyst/src/test/scala/org/apache/spark/sql/connector/expressions/TransformExtractorSuite.scala [hunk 2]
```diff
@@ -130,23 +130,32 @@ class TransformExtractorSuite extends SparkFunSuite {
...
-    }
+    // CLUSTERED BY (a, b)
+    checkBucketTransform(Seq(ref("a"), ref("b")), Seq("a", "b"))
+    // CLUSTERED BY (a.b)
+    checkBucketTransform(Seq(ref("a", "b")), Seq("a.b"))
+    checkBucketTransform(Seq(ref("a.b")), Seq("a.b"))
+    checkBucketTransform(Seq(ref("`a.b`")), Seq("`a.b`"))
 
-    transform("unknown", ref("a", "b")) match {
+    transform("unknown", ref("c")) match {
```
Comment (rdblue) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/connector/expressions/TransformExtractorSuite.scala:158]: Why did this test case change?

**Pull request closed without merging**